### DISCUSSION TITLE
Refine search results by HubSite (DepartmentId) for FilterWebpart

### DISCUSSION THREAD

**Discussion by MarcoSPDeveloper** [microsoft-search/pnp-modern-search#2459] [Q&A]
My users have a requirement for the filter web part (v4)

As a pnp modern search user i want to refine the search results by Hub site (DepartmentId). The refiner should show the hub site title as option. Expeced result: the user sees only search result from the selected hub site.

Can someone give me a hint how I can implement this as a developer?
Is there a possibility to set this filter option.

In gratitude
Marco

**Comment by wobba** [comment]
Query: `RelatedHubSites:<hubid>`
Refiner: map ows_SiteName to e.g. RefinableString00 and use that. Will require re-indexing.

Also see https://microsoft-search.github.io/pnp-modern-search/usage/search-results/tokens/#site-web-hub-etc-tokens on token usage to pick up the current hub.

**Reply by MarcoSPDeveloper** [reply]
Hi @wobba,

thanks for the Answer. Unfortunately that doesn't quite fit
ows_SiteName contains only the Title of the Site Collection, but i want to filter by Hub Site Collection Title.

Example:
If want to see Hub 1 and Hub 2 as refiner. This are my 2 Hub Sites.

My Hub Site Structure is:
Hub 1
 - Subsite 1
 - Subsite 2
Hub 2
 - Subsite 3
 - Subsite 4

There is a property DepartmentID available.

**Reply by wobba** [reply]
Happy development 🙂 you should create a custom filter component/webpart with the logic you need.
